src/libfetchers/git.cc

```diff
@@ -435,7 +435,16 @@ struct GitInputScheme : InputScheme
         //
         // See: https://discourse.nixos.org/t/57783 and #9708
         //
-        repoInfo.url = repoInfo.isLocal ? std::filesystem::absolute(url.path).string() : url.to_string();
+        if (repoInfo.isLocal) {
+            if (!isAbsolute(url.path)) {
+                warn(
+                    "Fetching Git repository '%s', which uses a path relative to the current directory. "
+                    "This is not supported and will stop working in a future release.",
+                    url);
+            }
+            repoInfo.url = std::filesystem::absolute(url.path).string();
```

Comment:
Is this still necessary after #12254? If not, we can simply drop it. The `absolute(url.path)` fix was proposed before #12254.

Reply:
Yes, it's still needed. Only CLI arguments are absolutized.